sdk/nodejs/output.ts

```diff
@@ -444,6 +494,73 @@ function getResourcesAndDetails<T>(allOutputs: Output<Unwrap<T>>[]): [Resource[]
     return [allResources, isKnown, isSecret];
 }
 
+/**
+ * Unknown represents a value that is unknown. These values correspond to unknown property values received from the
+ * Pulumi engine as part of the result of a resource registration (see runtime/rpc.ts). User code is not typically
+ * exposed to these values: any Output<> that contains an Unknown will itself be unknown, so any user callbacks
+ * passed to `apply` will not be run. Internal callers of `apply` can request that they are run even with unknown
+ * values; the output proxy takes advantage of this to allow proxied property accesses to return known values even
+ * if other properties of the containing object are unknown.
+ */
+class Unknown {
+     /**
+     * @internal
+     * A private field to help with RTTI that works in SxS scenarios.
+     *
+     * This is internal instead of being truly private, to support mixins and our serialization model.
+     */
+    // tslint:disable-next-line:variable-name
+    public readonly __pulumiUnknown: boolean = true;
+
+    /**
+     * Returns true if the given object is an instance of Unknown. This is designed to work even when
+     * multiple copies of the Pulumi SDK have been loaded into the same process.
+     */
+    public static isInstance(obj: any): obj is Unknown {
+        return utils.isInstance(obj, "__pulumiUnknown");
```

Comment:
technically you want to do `utils.isInstance<Unknown>`.  That ensures that ts will typecheck and validate that the string you passed in matches the name of the property.

Reply:
Why don't we do the same for `OutputImpl`?

Reply:
because i'm fallible :D

Reply:
can you still make this change? :)